Add tips and good practice for plugins developers (#6909)
qgis/QGIS-Documentation: docs/pyqgis_developer_cookbook/plugins/plugins.rst

Don't forget to set :class:`QAction` and :class:`QMenu` ``objectName`` to a name
specific to your plugin so that it can be customized.

> If your plugin only implements a processing provider and does not need

Reviewer: You can add the help even if your plugin have a UI, so I would not specify this sentence, no ?

Author: I'm not sure I understand what you mean: remove the whole sentence or one of the conditions? The idea was to avoid that plugins that do not need additional menu to work (eg under Plugins menu) create one just for their help/about button. If you already have a menu, it would look more coherent to me to put your help file in it.
Maybe, if you have a rewording suggestion...:wink:

Reviewer: > If you already have a menu, it would look more coherent to me to put your help file in it.

I see what you mean. Not a strong opinion. I took the habit to provide this shortcut in `iface.pluginHelpMenu()` whatever the plugin, to centralize all links to help from plugins.

Some plugins might neither have a menu nor a processing provider, for instance "French Locator Filter" for the French BAN. It's located only in the locator bar.

Author: OK, I'll try to reword and just inform people that the possibility exists...

> additional ui menu, you are encouraged to insert your help and about actions in

Reviewer: ```suggestion
additional UI menu, you are encouraged to insert your help and about actions in
```

> If your plugin only implements a processing provider and does not need
> additional ui menu, you are encouraged to insert your help and about actions in
> QGIS main :menuselection:`Help --> Plugins` menu. This can be done using the

Author: ```suggestion
While help and about actions can also be added to your custom menu,
a convenient place to make them available is in the 
QGIS main :menuselection:`Help --> Plugins` menu. This is done using the
```

:meth:`pluginHelpMenu() <qgis.gui.QgisInterface.pluginHelpMenu>` method.

.. testcode:: plugins

    def initGui(self):

        self.help_action = QAction(
            QIcon(":/plugins/testplug/icon.png"),
            self.tr("Test Plugin..."),
            self.iface.mainWindow()
        )
        # Add the action to the Help menu
        self.iface.pluginHelpMenu().addAction(self.help_action)

>         self.help_action.triggered.connect(self.showHelp)

Reviewer: ```suggestion
        self.help_action.triggered.connect(self.show_help)
        
    @staticmethod
    def show_help():
        """ Open the online help. """
        QDesktopServices.openUrl(QUrl('https://docs.qgis.org'))
```
Minor, let's use Python convention.
Other `initGui` inherited from QGIS, so it can't follow python convention.
